OutOfSchool/OutOfSchool.BusinessLogic/Services/Database/WorkshopService.cs

```diff
@@ -631,6 +648,132 @@ public async Task<SearchResult<WorkshopCard>> GetNearestByFilter(WorkshopFilter
         return result;
     }
 
+    /// <inheritdoc/>
+    public async Task<SearchResult<WorkshopDto>> FetchByFilterForAdmins(WorkshopFilterAdministration filter = null)
+    {
+        logger.LogInformation("Started retrieving Workshops by filter for admins.");
+
+        if (filter == null)
+        {
+            logger.LogDebug("Method {MethodName} started with null filter. Applying default {Filter}", nameof(FetchByFilterForAdmins), nameof(WorkshopFilterAdministration));
+            filter = new WorkshopFilterAdministration();
+        }
+
+        var predicate = await PredicateBuildForAdminds(filter);
+
+        var workshops = await workshopRepository.Get(
+                skip: filter.From,
+                take: filter.Size,
+                includeProperties: includingPropertiesForMappingDtoModel,
+                whereExpression: predicate,
+                asNoTracking: true)
+            .ToListAsync()
+            .ConfigureAwait(false);
+
+        var workshopsCount = workshops.Count;
+        logger.LogInformation("Retrieved {WorkshopsCount} matching recods by filter for admins.", workshopsCount);
+
+        var workshopsDTO = mapper.Map<List<WorkshopDto>>(workshops);
+
+        var result = new SearchResult<WorkshopDto>()
+        {
+            TotalAmount = workshopsCount,
+            Entities = workshopsDTO,
+        };
+
+        return result;
+    }
+
+    private async Task<Expression<Func<Workshop, bool>>> PredicateBuildForAdminds(WorkshopFilterAdministration filter)
+    {
+        var predicate = PredicateBuilder.True<Workshop>();
+
+        // TODO: add condition for united territorial community admin
+        if (currentUserService.IsMinistryAdmin())
+        {
+            await ApplyFilterForMinistryAdmin(filter).
+                ConfigureAwait(false);
```

Comment:
Please check if it's necessary to change this code.

Right now `predicate` is not changed.
I believe it should (like in next `if`: `predicate = predicate.And(tempPredicate)`, where `tempPredicate` is result of `ApplyFilterForMinistryAdmin`)

Reply:
Wrong alert, now I see that `filter.InstitutionId` is changed inside the `ApplyFilterForMinistryAdmin` method. Not very intuitive code, but let it be as is